### DISCUSSION TITLE
Query notifications by delivery method

### DISCUSSION THREAD

**Discussion by smarquez1** [excid3/noticed#508] [Q&A]
Hello,

I'm using the Noticed gem and would like to query notifications that have been sent using a specific delivered_by method (e.g., email, Slack, SMS).

Is there a recommended way to retrieve notifications that have been delivered by a particular method?
For example:

How can I identify which notifications were delivered via email?
Is there built-in support for this, or should I directly query the database tables?
Any guidance or examples would be greatly appreciated.

Thank you!

**Comment by excid3** [comment]
We don't currently track the deliveries themselves, just the notification and recipients.

Probably the best way would be to add an `after_deliver` callback to `Noticed::DeliveryMethod` to catch all deliveries. https://github.com/excid3/noticed/blob/main/lib/noticed/delivery_method.rb#L31

**Reply by smarquez1** [reply]
Thanks for getting back at me Chris!

In order to use `after_deliver`, do I need to create a new `DeliveryMethod`?

I've been trying to add an `after_deliver` callback to my Event, so I can update a `noticed_notifications` new column without success.

This is what i'd like to achieve, but I can't figure it out.

```ruby
class MessagePushNotifier < Noticed::Event
  deliver_by :ios do |config|
    ...
  end

  after_deliver do
    notifications.last.update(kind: config.method)
  end
end
```
